scipy/stats/_continuous_distns.py

```diff
@@ -6294,6 +6294,161 @@ def _stats(self, a):
+
+            func = func_fixed_shape if fshape else func_all_free
+
+            s_max = np.max(data) - np.min(data)
+            u_min = np.min(data) - s_max
+            u_max = (1-np.finfo(float).eps)*np.min(data)
+
+            # for some a > 1 cases, `u_max` needs to be decreased even more
+            # s.t. `func(u_min)` and `func(u_max)` have opposite signs.
+            for i in range(1, 30):
+                # sometimes the inner calculations raise a TypeError due to
+                # an invalid result.
+                try:
+                    a = func(u_min, None)
+                    b = func(u_max, None)
+                    # if `a` and `b` are not of opposite sign, decrease `u_max`
+                    # by an increasing amount each interation.
+                    if np.sign(a) == np.sign(b):
+                        u_max = u_max - np.finfo(float).eps**(1/i)
+                    else:
+                        break
+                except TypeError:
+                    break
+
+            try:
+                floc = optimize.root_scalar(func, bracket=[u_min, u_max],
+                                            args=(fshape,)).root
+                fscale = get_s(data, floc)
+                fshape = fshape if fshape else get_a(floc, fscale, data)
+            except ValueError:
+                # catch the ValueError due to f(a) and f(b) not have different
+                # signs
+                fshape, floc, fscale = None, None, None
+            return fshape, floc, fscale
+
+        # Analytical formulas to be used under the assumption that `fshape < 1`
+        def get_s_a_lt(data, *args):
+            return np.max(data) - np.min(data)
+
+        def get_u_a_lt(data, *args):
+            return np.min(data)
+
+        def fixed_shape_only(data, get_a, get_u, get_s, fshape):
+            floc = get_u_a_lt(data)
+            fscale = get_s_a_lt(data)
+            return fshape, floc, fscale
+
+        def all_free_lt(data, get_a, get_u, get_s):
+            floc = get_u_a_lt(data)
+            fscale = get_s_a_lt(data)
+            fshape = get_a(floc, fscale, data)
+            return fshape, floc, fscale
+
+        # a generic fitting routing that allows for parametrization of
+        # the methods to determine the shape, location, and scale.
+        def fit_with_multiple_funcs(fshape, floc, fscale, data, get_u, get_s,
+                                    fixed_shape_only, all_free):
+            if floc is not None:
+                # `floc` is fixed, `fshape` and `fscale` may be deterermined
+                # through analytical equations
+                if floc > np.min(data):
+                    # a constraint to fitting is that the location < np.min
+                    raise ValueError("Invalid values in `data`.  Maximum"
+                                     " likelihood estimation with 'powerlaw'"
+                                     " requires loc < min(data).")
```

Comment:
I decided to use ValueError here rather than FitDataError because I think that it provides easier information to help the user fix the problem. 

FitDataError would say:

```
FitDataError: Invalid values in `data`.  Maximum likelihood 
estimation with stats.powerlaw requires that 0 < (x - loc)/scale  < 1 for
each x in `data`.
```

And even though that is true, and ` loc < min(data)` is derived from it, it doesn't really show what the exact issue was.